Sources/SwiftInspectorVisitors/Tests/TypeDescriptionSpec.swift

```diff
@@ -549,6 +630,128 @@ final class TypeDescriptionSpec: QuickSpec {
+            """
+
+            visitor = MetatypeTypeSyntaxVisitor()
+            try? visitor.walkContent(content)
+          }
+
+          it("Finds the type") {
+            expect(visitor.metatypeTypeIdentifier?.asSource) == "Int.Type"
+          }
+        }
+
+        context("that represents a protocol") {
+          beforeEach {
+            let content = """
+            let metatype: Equatable.Protocol
+            """
+
+            visitor = MetatypeTypeSyntaxVisitor()
+            try? visitor.walkContent(content)
+          }
+
+          it("Finds the type") {
+            expect(visitor.metatypeTypeIdentifier?.asSource) == "Equatable.Protocol"
+          }
+        }
+      }
+
+      context("when called on a TypeSyntax node representing a SomeTypeSyntax") {
+        final class SomeTypeSyntaxVisitor: SyntaxVisitor {
+          var someTypeIdentifier: TypeDescription?
+          override func visit(_ node: SomeTypeSyntax) -> SyntaxVisitorContinueKind {
+            someTypeIdentifier = TypeSyntax(node).typeDescription
+            return .skipChildren
+          }
+        }
+
+        var visitor: SomeTypeSyntaxVisitor!
+        beforeEach {
+          let content = """
+            func makeSomething() -> some Equatable { "" }
+            """
+
+          visitor = SomeTypeSyntaxVisitor()
+          try? visitor.walkContent(content)
+        }
+
+        it("Finds the type") {
+          expect(visitor.someTypeIdentifier?.asSource) == "some Equatable"
+        }
+      }
+
+      context("when called on a TypeSyntax node representing an AttributedTypeSyntax") {
+        final class AttributedTypeSyntaxVisitor: SyntaxVisitor {
+          var attributedTypeIdentifier: TypeDescription?
+          override func visit(_ node: AttributedTypeSyntax) -> SyntaxVisitorContinueKind {
+            attributedTypeIdentifier = TypeSyntax(node).typeDescription
+            return .skipChildren
+          }
+        }
+
+        var visitor: AttributedTypeSyntaxVisitor!
+        context("with a specifier") {
+          beforeEach {
+            let content = """
+            inout Int
+            """
+
+            visitor = AttributedTypeSyntaxVisitor()
+            try? visitor.walkContent(content)
+          }
+
+          it("Finds the type") {
+            expect(visitor.attributedTypeIdentifier?.asSource) == "inout Int"
+          }
+        }
+
+        context("with attributes") {
+          beforeEach {
+            let content = """
+            @autoclosure () -> Void
+            """
+
+            visitor = AttributedTypeSyntaxVisitor()
+            try? visitor.walkContent(content)
+          }
+
+          it("Finds the type") {
+            expect(visitor.attributedTypeIdentifier?.asSource) == "@autoclosure () -> Void"
+          }
+
+          context("with a specifier and attributes") {
+            beforeEach {
+              let content = """
+              // This code doesn't compile but it can be parsed.
```

Comment:
as mentioned elsewhere I do not know if there exists compiling syntax that declares an attributed type with both a specifier and attributes